GPUP: Remove RemoteNativeImageBackendProxy, it is not useful

kkinnunen-apple opened the pull request:
#### b18ff101db00424e1d0c81442a55404c3a0e1ec3
<pre>
GPUP: Remove RemoteNativeImageBackendProxy, it is not useful
<a href="https://bugs.webkit.org/show_bug.cgi?id=299282">https://bugs.webkit.org/show_bug.cgi?id=299282</a>
<a href="https://rdar.apple.com/problem/161086182">rdar://problem/161086182</a>

Reviewed by Matt Woodrow.

RemoteNativeImageBackendProxy, it was overly complex for what it did.
The shareable bitmap can be held in RemoteResourceCacheProxy
over GPUP restarts. Later on this state will be moved to shared resource
cache.

This prevents the code from having overly many remote related concepts.
Later commits will introduce distinct type of NativeImage,
RemoteNativeImageProxy, that enables NativeImages that do not
immediately read their contents to WCP.

* Source/WebCore/platform/graphics/NativeImage.cpp:
(WebCore::NativeImage::create):
(WebCore::NativeImage::NativeImage):
(WebCore::NativeImage::~NativeImage):
(WebCore::NativeImage::platformImage const):
(WebCore::NativeImage::replacePlatformImage):
(WebCore::NativeImageBackend::isRemoteNativeImageBackendProxy const): Deleted.
(WebCore::PlatformImageNativeImageBackend::platformImage const): Deleted.
(WebCore::PlatformImageNativeImageBackend::PlatformImageNativeImageBackend): Deleted.
(WebCore::NativeImage::size const): Deleted.
(WebCore::NativeImage::hasAlpha const): Deleted.
(WebCore::NativeImage::colorSpace const): Deleted.
(WebCore::NativeImage::headroom const): Deleted.
(WebCore::NativeImage::replaceBackend): Deleted.
* Source/WebCore/platform/graphics/NativeImage.h:
(isType): Deleted.
* Source/WebCore/platform/graphics/RenderingResource.h:
(WebCore::RenderingResource::isNativeImage const): Deleted.
* Source/WebCore/platform/graphics/cairo/NativeImageCairo.cpp:
(WebCore::NativeImage::size const):
(WebCore::NativeImage::hasAlpha const):
(WebCore::NativeImage::colorSpace const):
(WebCore::NativeImage::headroom const):
(WebCore::PlatformImageNativeImageBackend::size const): Deleted.
(WebCore::PlatformImageNativeImageBackend::hasAlpha const): Deleted.
(WebCore::PlatformImageNativeImageBackend::colorSpace const): Deleted.
(WebCore::PlatformImageNativeImageBackend::headroom const): Deleted.
* Source/WebCore/platform/graphics/cg/NativeImageCG.cpp:
(WebCore::NativeImage::create):
(WebCore::NativeImage::size const):
(WebCore::NativeImage::hasAlpha const):
(WebCore::NativeImage::colorSpace const):
(WebCore::NativeImage::headroom const):
(WebCore::PlatformImageNativeImageBackend::size const): Deleted.
(WebCore::PlatformImageNativeImageBackend::hasAlpha const): Deleted.
(WebCore::PlatformImageNativeImageBackend::colorSpace const): Deleted.
(WebCore::PlatformImageNativeImageBackend::headroom const): Deleted.
* Source/WebCore/platform/graphics/skia/NativeImageSkia.cpp:
(WebCore::NativeImage::size const):
(WebCore::NativeImage::hasAlpha const):
(WebCore::NativeImage::colorSpace const):
(WebCore::NativeImage::headroom const):
(WebCore::PlatformImageNativeImageBackend::size const): Deleted.
(WebCore::PlatformImageNativeImageBackend::hasAlpha const): Deleted.
(WebCore::PlatformImageNativeImageBackend::colorSpace const): Deleted.
(WebCore::PlatformImageNativeImageBackend::headroom const): Deleted.
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/RemoteNativeImageBackendProxy.cpp: Removed.
* Source/WebKit/WebProcess/GPU/graphics/RemoteNativeImageBackendProxy.h: Removed.
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::didClose):
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.cpp:
(WebKit::createShareableBitmapForNativeImage):
(WebKit::RemoteResourceCacheProxy::recordNativeImageUse):
(WebKit::RemoteResourceCacheProxy::willDestroyNativeImage):
(WebKit::RemoteResourceCacheProxy::disconnect):
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:

Canonical link: <a href="https://commits.webkit.org/300463@main">https://commits.webkit.org/300463@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29923ea79a1358bb3e4c46fa26c88fa85c079397

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122661 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42373 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33063 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129277 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/74765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1dfbca16-6f85-43d7-8bd9-7adb4db3a87a) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/124537 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43092 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50966 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/93227 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/74765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d289f03f-1933-4921-8ef3-9820435a4e8a) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125613 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34353 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109809 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73871 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f6ed024d-6875-4217-a573-9e56a7af9e3f) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33340 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27965 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72762 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104050 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28176 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132002 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/49606 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37752 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/101760 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49982 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106025 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/101625 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25799 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46991 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25160 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/46376 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/49463 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55216 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/48930 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/52282 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50613 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/31f38ed4f4b11acf74445769772d2c5e5393ad04. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51110

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/bf8b719bcf1852be436939074324256071d59085. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51110

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/aac8ee380203e69f0c66bd362b5cd0b04b09e870. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51110

kkinnunen-apple committed:
GPUP: Remove RemoteNativeImageBackendProxy, it is not useful
https://bugs.webkit.org/show_bug.cgi?id=299282
rdar://problem/161086182

Reviewed by Matt Woodrow.

RemoteNativeImageBackendProxy, it was overly complex for what it did.
The shareable bitmap can be held in RemoteResourceCacheProxy
over GPUP restarts. Later on this state will be moved to shared resource
cache.

This prevents the code from having overly many remote related concepts.
Later commits will introduce distinct type of NativeImage,
RemoteNativeImageProxy, that enables NativeImages that do not
immediately read their contents to WCP.

* Source/WebCore/platform/graphics/NativeImage.cpp:
(WebCore::NativeImage::create):
(WebCore::NativeImage::NativeImage):
(WebCore::NativeImage::~NativeImage):
(WebCore::NativeImage::platformImage const):
(WebCore::NativeImage::replacePlatformImage):
(WebCore::NativeImageBackend::isRemoteNativeImageBackendProxy const): Deleted.
(WebCore::PlatformImageNativeImageBackend::platformImage const): Deleted.
(WebCore::PlatformImageNativeImageBackend::PlatformImageNativeImageBackend): Deleted.
(WebCore::NativeImage::size const): Deleted.
(WebCore::NativeImage::hasAlpha const): Deleted.
(WebCore::NativeImage::colorSpace const): Deleted.
(WebCore::NativeImage::headroom const): Deleted.
(WebCore::NativeImage::replaceBackend): Deleted.
* Source/WebCore/platform/graphics/NativeImage.h:
(isType): Deleted.
* Source/WebCore/platform/graphics/RenderingResource.h:
(WebCore::RenderingResource::isNativeImage const): Deleted.
* Source/WebCore/platform/graphics/cairo/NativeImageCairo.cpp:
(WebCore::NativeImage::size const):
(WebCore::NativeImage::hasAlpha const):
(WebCore::NativeImage::colorSpace const):
(WebCore::NativeImage::headroom const):
(WebCore::PlatformImageNativeImageBackend::size const): Deleted.
(WebCore::PlatformImageNativeImageBackend::hasAlpha const): Deleted.
(WebCore::PlatformImageNativeImageBackend::colorSpace const): Deleted.
(WebCore::PlatformImageNativeImageBackend::headroom const): Deleted.
* Source/WebCore/platform/graphics/cg/NativeImageCG.cpp:
(WebCore::NativeImage::create):
(WebCore::NativeImage::size const):
(WebCore::NativeImage::hasAlpha const):
(WebCore::NativeImage::colorSpace const):
(WebCore::NativeImage::headroom const):
(WebCore::PlatformImageNativeImageBackend::size const): Deleted.
(WebCore::PlatformImageNativeImageBackend::hasAlpha const): Deleted.
(WebCore::PlatformImageNativeImageBackend::colorSpace const): Deleted.
(WebCore::PlatformImageNativeImageBackend::headroom const): Deleted.
* Source/WebCore/platform/graphics/skia/NativeImageSkia.cpp:
(WebCore::NativeImage::size const):
(WebCore::NativeImage::hasAlpha const):
(WebCore::NativeImage::colorSpace const):
(WebCore::NativeImage::headroom const):
(WebCore::PlatformImageNativeImageBackend::size const): Deleted.
(WebCore::PlatformImageNativeImageBackend::hasAlpha const): Deleted.
(WebCore::PlatformImageNativeImageBackend::colorSpace const): Deleted.
(WebCore::PlatformImageNativeImageBackend::headroom const): Deleted.
* Source/WebKit/Sources.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/RemoteNativeImageBackendProxy.cpp: Removed.
* Source/WebKit/WebProcess/GPU/graphics/RemoteNativeImageBackendProxy.h: Removed.
* Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp:
(WebKit::RemoteRenderingBackendProxy::didClose):
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.cpp:
(WebKit::createShareableBitmapForNativeImage):
(WebKit::RemoteResourceCacheProxy::recordNativeImageUse):
(WebKit::RemoteResourceCacheProxy::willDestroyNativeImage):
(WebKit::RemoteResourceCacheProxy::disconnect):
* Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:

Canonical link: https://commits.webkit.org/300463@main

webkit-commit-queue commented:
Committed 300463@main (b18ff101db00): <https://commits.webkit.org/300463@main>

Reviewed commits have been landed. Closing PR #51110 and removing active labels.
